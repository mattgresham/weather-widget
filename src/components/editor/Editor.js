// @flow
import React, { Component, Fragment } from 'react';
import Input from '../input';
import InputWrapper from '../input-wrapper';
import RadioButtonGroup from '../radio-button-group';
import temperatureUnitOptions from '../../constants/temperatureUnits';
import windOptions from '../../constants/wind';
import type { Units } from '../../types/units';

type Props = {
  isWindShown: string,
  setIsWindShown: Function,
  setTitle: Function,
  setUnits: Function,
  title: string,
  units: Units
};

class Editor extends Component<Props> {
  onTitleChange = (e: SyntheticEvent<HTMLInputElement>) => {
    this.props.setTitle(e.currentTarget.value);
  };

  onUnitsChange = (e: SyntheticEvent<HTMLInputElement>) => {
    this.props.setUnits(e.currentTarget.value);
  };

  onIsWindShownChange = (e: SyntheticEvent<HTMLInputElement>) => {
    this.props.setIsWindShown(e.currentTarget.value);
  };

  render() {
    const { isWindShown, title, units } = this.props;

    return (
      <Fragment>
        <InputWrapper title="Title">
          <Input
            onChange={this.onTitleChange}
            placeholder="Title of widget"
            value={title}
          />
        </InputWrapper>
        <InputWrapper title="Temperature">
          <RadioButtonGroup
            onChange={this.onUnitsChange}
            value={units}
            values={temperatureUnitOptions}
          />
        </InputWrapper>
        <InputWrapper title="Wind">
          <RadioButtonGroup
            onChange={this.onIsWindShownChange}
            value={isWindShown}
            values={windOptions}
          />
        </InputWrapper>
      </Fragment>
    );
  }
}

export default Editor;
