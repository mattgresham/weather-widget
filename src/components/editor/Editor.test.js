import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Editor from './Editor';

const mockFn = jest.fn();

it('renders correctly', () => {
  const component = render(
    <Editor
      isWindShown="on"
      setIsWindShown={mockFn}
      setTitle={mockFn}
      setUnits={mockFn}
      title=""
      units="metric"
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('renders correctly with a title', () => {
  const component = render(
    <Editor
      isWindShown="on"
      setIsWindShown={mockFn}
      setTitle={mockFn}
      setUnits={mockFn}
      title="Title"
      units="metric"
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});
