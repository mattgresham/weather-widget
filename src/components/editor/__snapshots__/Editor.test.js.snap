// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    class="container"
  >
    <div
      class="title"
    >
      Title
    </div>
    <input
      class="element"
      placeholder="Title of widget"
      type="text"
      value=""
    />
  </div>,
  <div
    class="container"
  >
    <div
      class="title"
    >
      Temperature
    </div>
    <div
      class="group"
    >
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__metric"
        >
          <input
            checked=""
            class="element"
            id="radio-button__metric"
            type="radio"
            value="metric"
          />
          째C
        </label>
      </div>
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__imperial"
        >
          <input
            class="element"
            id="radio-button__imperial"
            type="radio"
            value="imperial"
          />
          째F
        </label>
      </div>
    </div>
  </div>,
  <div
    class="container"
  >
    <div
      class="title"
    >
      Wind
    </div>
    <div
      class="group"
    >
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__on"
        >
          <input
            checked=""
            class="element"
            id="radio-button__on"
            type="radio"
            value="on"
          />
          On
        </label>
      </div>
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__off"
        >
          <input
            class="element"
            id="radio-button__off"
            type="radio"
            value="off"
          />
          Off
        </label>
      </div>
    </div>
  </div>,
]
`;

exports[`renders correctly with a title 1`] = `
Array [
  <div
    class="container"
  >
    <div
      class="title"
    >
      Title
    </div>
    <input
      class="element"
      placeholder="Title of widget"
      type="text"
      value="Title"
    />
  </div>,
  <div
    class="container"
  >
    <div
      class="title"
    >
      Temperature
    </div>
    <div
      class="group"
    >
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__metric"
        >
          <input
            checked=""
            class="element"
            id="radio-button__metric"
            type="radio"
            value="metric"
          />
          째C
        </label>
      </div>
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__imperial"
        >
          <input
            class="element"
            id="radio-button__imperial"
            type="radio"
            value="imperial"
          />
          째F
        </label>
      </div>
    </div>
  </div>,
  <div
    class="container"
  >
    <div
      class="title"
    >
      Wind
    </div>
    <div
      class="group"
    >
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__on"
        >
          <input
            checked=""
            class="element"
            id="radio-button__on"
            type="radio"
            value="on"
          />
          On
        </label>
      </div>
      <div
        class="container"
      >
        <label
          class="label"
          for="radio-button__off"
        >
          <input
            class="element"
            id="radio-button__off"
            type="radio"
            value="off"
          />
          Off
        </label>
      </div>
    </div>
  </div>,
]
`;
